### PR TITLE
Extract cpp_rsc

### PR DESCRIPTION
**Pull request by JonasVautherin**
Not sure how ugly this is, but:

* It may enable the camera definition files for Windows
* It should improve cross-compilation, as the `camera_definition_files.*` are now versioned
* It removes some conditional code in the CMakeLists
* It removes `#ifdefs` in the code

@julianoes: What do you think? I'm mainly interested in seeing if that fixes the dockcross compilation after the protoc PR is merged.

### CODE REVIEW

**Comment by JonasVautherin** [comment]
Note: together with #839, this allows me to build for android-arm64 with dockcross (and I expect it to work for other dockcross configurations).

File: src/plugins/camera/camera_definition_files/generated/camera_definition_files.cpp [hunk 1]
```diff
@@ -0,0 +1,13791 @@
+/* Automatic generated file. DO NOT EDIT.
+   The file was generated by cpp_rsc program
+   https://github.com/orex/cpp_rsc
+   Resource file name: /tmp/tmp.FanET5d6hS/camera_definition_config.rsc
+   This is a source file*/
+
+#include "camera_definition_files.h"
+
+  /* Variable cgoetxml from file /home/jones/Documents/px4/sdk/mavsdk/tools/../src/plugins/camera/camera_definition_files/cgoet.xml */
```
Comment (julianoes) [inline src/plugins/camera/camera_definition_files/generated/camera_definition_files.cpp]: It would be better if the path doesn't end up in here :thinking:

Comment (JonasVautherin) [reply src/plugins/camera/camera_definition_files/generated/camera_definition_files.cpp]: Yeah I was not a big fan of those comments, but I did not really feel like patching it :sweat_smile:. Anyway, I forked it and removed the comments!

Comment (julianoes) [reply src/plugins/camera/camera_definition_files/generated/camera_definition_files.cpp]: Oh really, nice :)

File: src/plugins/camera/camera_definition_files/generated/camera_definition_files.cpp [hunk 2]
```diff
@@ -0,0 +1,13786 @@
...
+  '\x22', '\x20', '\x65', '\x6e', '\x63', '\x6f', '\x64', '\x69', '\x6e', 
+  '\x67', '\x3d', '\x22', '\x55', '\x54', '\x46', '\x2d', '\x38', '\x22', 
+  '\x20', '\x3f', '\x3e', '\x0a', '\x3c', '\x6d', '\x61', '\x76', '\x6c', 
+  '\x69', '\x6e', '\x6b', '\x63', '\x61', '\x6d', '\x65', '\x72', '\x61', 
+  '\x3e', '\x0a', '\x20', '\x20', '\x20', '\x20', '\x3c', '\x64', '\x65', 
+  '\x66', '\x69', '\x6e', '\x69', '\x74', '\x69', '\x6f', '\x6e', '\x20', 
+  '\x76', '\x65', '\x72', '\x73', '\x69', '\x6f', '\x6e', '\x3d', '\x22', 
+  '\x36', '\x22', '\x3e', '\x0a', '\x20', '\x20', '\x20', '\x20', '\x20', 
+  '\x20', '\x20', '\x20', '\x3c', '\x6d', '\x6f', '\x64', '\x65', '\x6c', 
+  '\x3e', '\x43', '\x47', '\x4f', '\x45', '\x54', '\x3c', '\x2f', '\x6d',
```
Comment (julianoes) [inline src/plugins/camera/camera_definition_files/generated/camera_definition_files.cpp]:
```suggestion
  '\x3e', '\x42', '\x47', '\x4f', '\x45', '\x54', '\x3c', '\x2f', '\x6d', 
```
Pretty sure! :smile:

**Pull request merged**